interactors/nonceHandlerV2/nonceTransactionsHandler.go

```diff
@@ -28,30 +35,35 @@ var log = logger.GetOrCreate("elrond-sdk-erdgo/interactors/nonceHandlerV2")
 type nonceTransactionsHandlerV2 struct {
 	proxy            interactors.Proxy
 	mutHandlers      sync.RWMutex
-	handlers         map[string]*addressNonceHandler
+	creator          interactors.AddressNonceHandlerCreator
+	handlers         map[string]interactors.AddressNonceHandler
 	cancelFunc       func()
 	intervalToResend time.Duration
 }
 
 // NewNonceTransactionHandlerV2 will create a new instance of the nonceTransactionsHandlerV2. It requires a Proxy implementation
 // and an interval at which the transactions sent are rechecked and eventually, resent.
-func NewNonceTransactionHandlerV2(proxy interactors.Proxy, intervalToResend time.Duration) (*nonceTransactionsHandlerV2, error) {
-	if check.IfNil(proxy) {
+func NewNonceTransactionHandlerV2(args ArgsNonceTransactionsHandlerV2) (*nonceTransactionsHandlerV2, error) {
+	if check.IfNil(args.Proxy) {
 		return nil, interactors.ErrNilProxy
 	}
-	if intervalToResend < minimumIntervalToResend {
+	if args.IntervalToResend < minimumIntervalToResend {
 		return nil, fmt.Errorf("%w for intervalToResend in NewNonceTransactionHandlerV2", interactors.ErrInvalidValue)
 	}
+	if check.IfNil(args.Creator) {
+		return nil, interactors.ErrNilAddressNonceHandlerCreator
+	}
 
 	nth := &nonceTransactionsHandlerV2{
-		proxy:            proxy,
-		handlers:         make(map[string]*addressNonceHandler),
-		intervalToResend: intervalToResend,
+		proxy:            args.Proxy,
+		handlers:         make(map[string]interactors.AddressNonceHandler),
+		intervalToResend: args.IntervalToResend,
+		creator:          args.Creator,
 	}
 
 	ctx, cancelFunc := context.WithCancel(context.Background())
 	nth.cancelFunc = cancelFunc
-	go nth.resendTransactionsLoop(ctx, intervalToResend)
+	go nth.resendTransactionsLoop(ctx, args.IntervalToResend)
```

Comment:
useless passing of the interval argument, on L152 & L160 could have used nth.intervalToResend